json-core/src/main/java/io/micronaut/json/convert/JsonConverterRegistrar.java

```diff
@@ -176,6 +177,9 @@ protected TypeConverter<Map, Object> mapToObjectConverter() {
             }
             ArgumentBinder binder = this.beanPropertyBinder.get();
             ArgumentBinder.BindingResult result = binder.bind(conversionContext, correctKeys(map));
+            conversionContext.getLastError().ifPresent(error -> {
+                throw new ConversionErrorException(conversionContext.getArgument(), error);
+            });
```

Comment:
this will modify universally for all type conversions that use this type converter which is a breaking change.

We need to modify only the place where the conversion happens to throw this error

Reply:
@n0tl3ss can you address @graemerocher PR feedback? thanks.

Reply:
@sdelamo Offline I told Graeme that I think this is the right place where I should expect error to be thrown. He told me that he will take a look how better to do it.